src/services/ethereum/CustomSigner.js

```diff
@@ -0,0 +1,104 @@
+// @flow
+
+import { Navigation } from 'react-native-navigation';
+import { screen } from '../../global/Screens';
+
+const ethers = require('ethers');
+
+/**
+ * @desc Custom signer for ethereum RPC
+ * @example
+ * const customSigner = CustomSigner(PRIVATE_KEY, 'homestead')
+ *
+ * @param {string} privateKey Private key of wallet
+ * @param {string} provider name of network
+ * @return {object} custom signer with wallet functions
+ */
+export default function CustomSigner(privateKey: string, provider: string) {
+  const wallet = new ethers.Wallet(privateKey);
+  // wallet.provider = new ethers.providers.InfuraProvider(provider);
+  this.provider = new ethers.providers.InfuraProvider(provider);
+  this.address = wallet.address;
+  this.getBalance = wallet.getBalance;
+  this.estimateGas = wallet.estimateGas;
+  this.getTransactionCount = wallet.getTransaction;
+  this.sign = async (transaction) => {
+    const transactionObject = transaction;
+    try {
+      const signedTransaction = await new Promise((resolve, reject) => {
+        Navigation.showModal({
+          ...screen('CONFIRMATION_SCREEN'),
+          passProps: {
+            onFail: () => {
+              reject();
+            },
+            onSuccess: (gasPrice) => {
+              // Here we have gasPrice to pass it somewhere later.
+              transactionObject.gasPrice = ethers.utils.bigNumberify(`${gasPrice.toString()}000000000`);
```

Comment:
This way of adding zeros to gasPrice looks pretty unstable, since I'm not sure how much digits is in gasPrice. It should at least be documented

Reply:
Added the documentation